File: 08-DesignPatterns-Java/Starter/Adapter/src/Ducks/DonaldDuck.java
package Ducks;

import Fly.NoneFlyStrategy;
import Quack.HumanQuackStrategy;

public class DonaldDuck extends Duck {
    public DonaldDuck() {
        super(new HumanQuackStrategy(), new NoneFlyStrategy());
    }

    @Override
    public void display() {
        System.out.println("Looks like taken from a comic book.");
    }
}

File: 08-DesignPatterns-CSharp/08-1-3-Strategy-Solution/Ducks/DonaldDuck.cs
namespace Strategy.Ducks
{
    using System;
    using Strategy.Fly;
    using Strategy.Quack;

    public sealed class DonaldDuck : Duck
    {
        public DonaldDuck()
            : base(new HumanQuackStrategy(), new NoneFlyStrategy())
        {
        }

        public override void Display() => Console.WriteLine("Looks like taken from a comic book.");
    }
}